File: snippets/go/webhook.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

func RegisterWebhook(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: register-webook
	if err := sdk.RegisterWebhook("https://your-nds-service.com/api/v1/notify?platform=ios&token=<PUSH_TOKEN>"); err != nil {
		log.Printf("Webhook register failed: %v", err)
	}
	// ANCHOR_END: register-webook
}

func UnregisterWebhook(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: unregister-webook
	if err := sdk.UnregisterWebhook(); err != nil {
		log.Printf("Webhook unregister failed: %v", err)
	}
	// ANCHOR_END: unregister-webook
}

File: snippets/rust/src/webhook.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_liquid::prelude::*;

async fn register_webhook(sdk: Arc<LiquidSdk>) -> Result<()> {
    // ANCHOR: register-webook
    sdk.register_webhook(
        "https://your-nds-service.com/api/v1/notify?platform=ios&token=<PUSH_TOKEN>".to_string(),
    )
    .await?;
    // ANCHOR_END: register-webook

    Ok(())
}

async fn unregister_webhook(sdk: Arc<LiquidSdk>) -> Result<()> {
    // ANCHOR: unregister-webook
    sdk.unregister_webhook().await?;
    // ANCHOR_END: unregister-webook

    Ok(())
}
